hefquin-pgconnector/src/main/java/se/liu/ida/hefquin/engine/wrappers/lpg/conn/Neo4jConnectionFactory.java

```diff
@@ -65,7 +117,8 @@ public List<TableRecord> execute( final String cypherQuery ) throws Neo4jExcepti
 				throw new Neo4jConnectionException(this, "Neo4j server could not be reached.", e);
 			}
 
-			if ( response.statusCode() != 200 ) {
+			System.err.println("Code: " + response.statusCode());
+			if ( ! ( response.statusCode() >= 200 && response.statusCode() < 300 ) ) {
```

Comment:
Is there a specific reason why you changed this from accepting only the status code 200? I mean, I understand that all 2xx codes indicate a correct response to the specific request. Yet, I would still think that any 2xx code other than 200 is not an expected outcome in this given, no?

Reply:
According to the Neo4J API documentation, a query can return `200 OK` or `201 Created` when executed over the HTTP API because it wraps the request in an implicit transaction (i.e., the `201` can indicate that the transaction was successfully created, not that data was inserted/changed). It also lists `202 Accepted` as a possible response code in some circumstances. See https://neo4j.com/docs/query-api/current/query/

On my dockerized Neo4J instance I only get 201. Would it be better to include only the codes 200, 201 and 202, rather than to capture all 2XX codes perhaps?

Reply:
> Would it be better to include only the codes 200, 201 and 202, rather than to capture all 2XX codes perhaps?

Yes, I think so. Also, please add a comment directly in the code that states what you wrote here (to make sure that, the next time we come across this part of the code base, we don't wonder again why we accept exactly these three status codes).